### PR TITLE
Handle the case where target contains nans in CatToNumTransform

### LINKED ISSUE #172: Handle the case when target_mean is nan in CatToNumTransform
Currently, it will just cast all the transformed values to nan

### PR DESCRIPTION
**Pull request by yiweny**
Fixes #172

### CODE REVIEW

File: torch_frame/transforms/cat_to_num_transform.py [hunk 1]
```diff
@@ -59,6 +59,11 @@ def _fit(
         else:
             num_classes = 2
             target = tf_train.y.unsqueeze(1)
+            mask = ~torch.isnan(target)
+            if torch.isnan(target).any():
```
Comment (zechengz) [inline torch_frame/transforms/cat_to_num_transform.py:63]: nit
```suggestion
            if (~mask).sum() > 0:
```

**Review by zechengz — APPROVED** [review]

File: torch_frame/transforms/cat_to_num_transform.py (hunk 1 above)
Comment (weihua916) [reply torch_frame/transforms/cat_to_num_transform.py:63]: (~mask).any()

File: test/transforms/cat_to_num_transform.py [hunk 2]
```diff
@@ -123,3 +123,17 @@ def test_ordered_target_statistics_encoder(task_type):
...
             out.col_names_dict[stype.numerical][:total_numerical_cols])
+
+
+def test_cat_to_num_transform_with_target_contains_nans():
+    num_rows = 10
+    dataset: Dataset = FakeDataset(num_rows=num_rows, with_nan=False,
+                                   stypes=[stype.numerical, stype.categorical],
+                                   task_type=TaskType.REGRESSION,
+                                   create_split=True)
+    dataset.df.at[0, 'target'] = np.nan
```
Comment (weihua916) [inline test/transforms/cat_to_num_transform.py:134]: Why don't we modify `FakeDataset` so that `with_nan=True` gives nan values in the target as well>

Comment (yiweny) [reply test/transforms/cat_to_num_transform.py:134]: Makes sense. I hope making this change doesn't affect other test cases. 🥹

**Pull request merged**